pkg/apis/work/v1alpha2/binding_types.go

```diff
@@ -319,6 +319,23 @@ type ResourceBindingStatus struct {
 	// AggregatedStatus represents status list of the resource running in each member cluster.
 	// +optional
 	AggregatedStatus []AggregatedStatusItem `json:"aggregatedStatus,omitempty"`
+
+	// FailoverHistory represents history of the previous failovers of this resource
+	// + optional
+	FailoverHistory []FailoverHistoryItem `json:"failoverHistory,omitempty"`
+}
+
+// FailoverHistoryItem represents either an application or cluster failover event that occurred in the past.
+type FailoverHistoryItem struct {
+	// FailoverTime represents the timestamp when the workload failed over.
+	// It is represented in RFC3339 form(like '2021-04-25T10:02:10Z') and is in UTC.
+	FailoverTime *metav1.Time `json:"failoverTime,omitempty"`
+
+	// OriginCluster denotes the name of the cluster from which the workload was failed over.
+	OriginCluster string `json:"originCluster,omitempty"`
+
+	// Reason denotes the reason why the workload failed over.
+	Reason string `json:"reason,omitempty"`
```

Comment:
So that each history item can provide enough context for us to figure out where the application was migrated.

In addition, we can also have the preserved state in this history item.

Reply:
Yeah, you are right. 
In fact, the demo at https://github.com/karmada-io/karmada/pull/5116#issuecomment-2379037424 does not use the history at all.